platform/linux-generic/include/odp_bitset.h

```diff
@@ -27,7 +27,12 @@
 /* Find a suitable data type that supports lock-free atomic operations */
 #if defined(__aarch64__) && defined(__SIZEOF_INT128__) && \
 	__SIZEOF_INT128__ == 16
-#define LOCKFREE16
+/* Todo: on thunder-x with gcc4.8 and gcc5 lockfree 16 code with -O0
+ * compiles to hung forever code. With -O2 everything works.
+ * Disable lock free function for now as temporary fix.
+ *
+ * #define LOCKFREE16
+ */
```

Comment:
This is fine as a temporary workaround, but a more comprehensive solution might want to make use of the GCC `__OPTIMIZE__` builtin which is set for all optimizing compile levels (_i.e._, anything other than `-O0`). See [here](https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html) for a description of it.